### PR TITLE
feat(display-name): convert to React

### PR DESCRIPTION
**Pull request by virtuacoplenny**
- Create a new React Component for displaying and updating display
  names on small videos.
- The updating of the Component is defined in the parent class
  SmallVideo, which children will get access to through prototype
  copying.
- Create a new actionType and middleware so name changes that occur
  in DisplayName can be propogated to outside redux.

### CODE REVIEW

File: react/features/display-name/components/DisplayName.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+        /**
+         * The participant's current display name.
+         */
+        displayName: React.PropTypes.string,
+
+        /**
+         * A string to append to the displayName, if provided.
+         */
+        displayNameSuffix: React.PropTypes.string,
```
Comment (yanas) [inline react/features/display-name/components/DisplayName.js]: I don't remember what was this about?

File: react/features/display-name/components/DisplayName.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+        /**
+         * The ID attribute to add to the component. Useful for global querying
+         * for the component by legacy components and torture tests.
+         */
+        idAttribute: React.PropTypes.string,
```
Comment (yanas) [inline react/features/display-name/components/DisplayName.js]: Is this how we name this one throughout react? I would name it elementId or refId.

File: react/features/display-name/components/DisplayName.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+             *
+             * type {boolean}
+             */
+            isEditing: false,
+
+            /**
+             * The display name to show in the input field while editing the
+             * known display name.
+             */
+            pendingDisplayName: ''
```
Comment (yanas) [inline react/features/display-name/components/DisplayName.js]: Is this the default value for the edit field? Why now call it something like editDefaultValue or similar. I was confused the pending word.

**Comment by yanas** [comment]
Note: Many of the changes are an intermediate state while some of the components lay in the old code, so I haven't focused on those, because they'll be removed anyway.
Question: previously we were using UIUtil.escapeHtml(displayName), do you handle this now? If so can you point me to the code or just explain how we prevent javascript injection. Thanks!

**Comment by virtuacoplenny** [comment]
I wasn't worried about XSS because I read jsx handles escaping. https://facebook.github.io/react/docs/introducing-jsx.html#jsx-prevents-injection-attacks

**Comment by yanas** [comment]
UPDATE: Just tested a 3 way conference call and changing the local participant name from the Required Display Name prompt didn't result in a change for the other participants. Changing from the local thumbnail edit works though.

File: react/features/display-name/components/DisplayName.js (hunk 1 above)
Comment (virtuacoplenny) [inline react/features/display-name/components/DisplayName.js]: This is the "(me)" that displays for the local name when a name the participant has a display name set.

File: react/features/display-name/components/DisplayName.js (hunk 2 above)
Comment (virtuacoplenny) [inline react/features/display-name/components/DisplayName.js]: I think generally you might want it to be just 'id' but I thought that might be confusing with participantID. I'll change it to elementId.

File: react/features/display-name/components/DisplayName.js (hunk 3 above)
Comment (virtuacoplenny) [inline react/features/display-name/components/DisplayName.js]: It's the current entered value in the edit field. It gets set to the participant's display name when editing is begun or is set as an empty string. I can see why the empty string then could be confusing.

**Pull request merged**